File: src/main/java/sample/controller/dialogs/search/PatternsDialog.java
package sample.controller.dialogs.search;

import javafx.scene.control.ContextMenu;
import javafx.scene.control.Dialog;
import javafx.scene.control.ScrollPane;
import javafx.scene.layout.TilePane;
import javafx.stage.Modality;
import javafx.stage.Window;
import sample.model.patterns.Pattern;

import java.util.List;
import java.util.Map;

public class PatternsDialog extends Dialog {
    public static PatternTile selected;
    public PatternsDialog(List<Pattern> patternsList, List<Map<String, String>> additionalInfo, ContextMenu menu) {
        super();
        super.setResizable(true);
        super.initModality(Modality.NONE);

        TilePane pane = new TilePane();
        pane.setHgap(5);
        pane.setVgap(5);
        pane.prefWidthProperty().bind(super.widthProperty());

        for (int i = 0; i < patternsList.size(); i++) {
            PatternTile patternTile = new PatternTile(patternsList.get(i),
                    patternsList.get(i).toString(), additionalInfo.get(i));
            patternTile.setOnContextMenuRequested(event -> {
                menu.show(patternTile, event.getScreenX(), event.getScreenY());
                selected = patternTile;
            });
            pane.getChildren().add(patternTile);
        }


        ScrollPane scrollPane = new ScrollPane(pane);
        scrollPane.setStyle("-fx-background-color: transparent;");  // No border
        scrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
        scrollPane.prefWidthProperty().bind(super.widthProperty());
        scrollPane.prefHeightProperty().bind(super.heightProperty());

        super.setWidth(300);
        super.setHeight(300);
        super.getDialogPane().setContent(scrollPane);

        // Allows closing with close button
        Window window = super.getDialogPane().getScene().getWindow();
        window.setOnCloseRequest(event -> window.hide());
    }
}

File: src/main/kotlin/application/controller/dialogs/search/PatternsDialog.kt
package application.controller.dialogs.search

import application.model.patterns.Pattern
import javafx.event.EventHandler
import javafx.scene.control.ContextMenu
import javafx.scene.control.Dialog
import javafx.scene.control.ScrollPane
import javafx.scene.input.ContextMenuEvent
import javafx.scene.layout.TilePane
import javafx.stage.Modality
import javafx.stage.WindowEvent

class PatternsDialog(patternsList: List<Pattern>, additionalInfo: List<Map<String, String>>, menu: ContextMenu) :
    Dialog<Any?>() {
    companion object {
        @JvmField
        var selected: PatternTile? = null
    }

    init {
        super.setResizable(true)
        super.initModality(Modality.NONE)

        val pane = TilePane()
        pane.hgap = 5.0
        pane.vgap = 5.0
        pane.prefWidthProperty().bind(super.widthProperty())

        for (i in patternsList.indices) {
            val patternTile = PatternTile(patternsList[i], patternsList[i].toString(), additionalInfo[i])
            patternTile.onContextMenuRequested = EventHandler { event: ContextMenuEvent ->
                menu.show(patternTile, event.screenX, event.screenY)
                selected = patternTile
            }
            pane.children.add(patternTile)
        }

        val scrollPane = ScrollPane(pane)
        scrollPane.style = "-fx-background-color: transparent;" // No border
        scrollPane.hbarPolicy = ScrollPane.ScrollBarPolicy.NEVER
        scrollPane.prefWidthProperty().bind(super.widthProperty())
        scrollPane.prefHeightProperty().bind(super.heightProperty())

        super.setWidth(300.0)
        super.setHeight(300.0)
        super.getDialogPane().content = scrollPane

        // Allows closing with close button
        val window = super.getDialogPane().scene.window
        window.onCloseRequest = EventHandler { event: WindowEvent? -> window.hide() }
    }
}